x-pack/plugins/alerting/server/task_runner/rule_action_helper.ts

```diff
@@ -10,15 +10,25 @@ import {
   IntervalSchedule,
   parseDuration,
   RuleAction,
+  RuleDefaultAction,
   RuleNotifyWhenTypeValues,
   ThrottledActions,
 } from '../../common';
+import { isSystemAction } from '../../common/system_actions/is_system_action';
 
 export const isSummaryAction = (action?: RuleAction) => {
+  if (action != null && isSystemAction(action)) {
```

Comment:
I though to distinguish between actions that support summarization and system action to have more control over system actions. A summary action is an action that defines `frequency?.summary` but a system action is something different.